notebooks/resource-allocation/generate-allocation-summary-arrays.py

```diff
@@ -0,0 +1,239 @@
+    rvals = StringIO(check_output(sacct + [jid]).decode('ascii'))
+    _d = pd.read_csv(rvals, sep='|')
+    jmem = _d.MaxRSS.apply(lambda x: x if type(x) is not str
+                           else MaxRSS_helper(x)).max()
+    jwt = _d.ElapsedRaw.max()
+
+    rvals = StringIO(check_output(sacct + [mjid]).decode('ascii'))
+    _d = pd.read_csv(rvals, sep='|')
+    mmem = _d.MaxRSS.apply(lambda x: x if type(x) is not str
+                           else MaxRSS_helper(x)).max()
+    mwt = _d.ElapsedRaw.max()
+
+    data.append({
+        'jid': j.id, 'sjid': jid, 'mem': jmem, 'wt': jwt, 'type': 'main',
+        'db': j.parameters.values['Database'].split('/')[-1]})
+    data.append(
+        {'jid': j.id, 'sjid': mjid, 'mem': mmem, 'wt': mwt, 'type': 'merge',
+         'db': j.parameters.values['Database'].split('/')[-1]})
+df = pd.DataFrame(data)
+df.to_csv(fn, sep='\t', index=False)
+
+# 2. qp-meta Sortmerna
+
+fn = join('/panfs', 'qiita', 'jobs_sortmerna.tsv.gz')
+print(f"Generating the summary for the woltka jobs: {fn}.")
+
+# for woltka we will only use jobs from the last 6 months
+cmds = [c for s in Software.iter(False)
+        if 'minimap2' in s.name.lower() for c in s.commands]
+jobs = [j for c in cmds for j in c.processing_jobs if j.status == 'success' and
+        j.heartbeat.date() > six_months and j.input_artifacts]
+
+data = []
+for j in jobs:
+    size = sum([fp['fp_size'] for fp in j.input_artifacts[0].filepaths])
+    jid, mjid = j.external_id.strip().split()
+    rvals = StringIO(check_output(sacct + [jid]).decode('ascii'))
+    _d = pd.read_csv(rvals, sep='|')
+    jmem = _d.MaxRSS.apply(lambda x: x if type(x) is not str
+                           else MaxRSS_helper(x)).max()
+    jwt = _d.ElapsedRaw.max()
+
+    rvals = StringIO(check_output(sacct + [mjid]).decode('ascii'))
+    _d = pd.read_csv(rvals, sep='|')
+    mmem = _d.MaxRSS.apply(lambda x: x if type(x) is not str
+                           else MaxRSS_helper(x)).max()
+    mwt = _d.ElapsedRaw.max()
+
+    data.append({
+        'jid': j.id, 'sjid': jid, 'mem': jmem, 'wt': jwt, 'type': 'main'})
+    data.append(
+        {'jid': j.id, 'sjid': mjid, 'mem': mmem, 'wt': mwt, 'type': 'merge'})
+df = pd.DataFrame(data)
+df.to_csv(fn, sep='\t', index=False)
+
+
+# 3. Adapter and host filtering. Note that there is a new version deployed on
+#    Jan 2024 so the current results will not be the most accurate
+
+fn = join('/panfs', 'qiita', 'jobs_adapter_host.tsv.gz')
+print(f"Generating the summary for the woltka jobs: {fn}.")
+
+# for woltka we will only use jobs from the last 6 months
+cmds = [c for s in Software.iter(False)
+        if 'meta' in s.name.lower() for c in s.commands]
+jobs = [j for c in cmds if 'sortmerna' in c.name.lower()
+        for j in c.processing_jobs if j.status == 'success' and
+        j.heartbeat.date() > six_months and j.input_artifacts]
+
+data = []
+for j in jobs:
+    size = sum([fp['fp_size'] for fp in j.input_artifacts[0].filepaths])
```

Comment:
This loop appears to be repeated several times. In the future we can generalize it into a function.